### ISSUE TITLE
[Bug]: txn control not applied to snapshot.Remove while meta gc

### ISSUE THREAD

**Issue by longjiquan** [milvus-io/milvus#29123]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

https://github.com/milvus-io/milvus/blob/3c32ba240784db5e8cda145d53e140b9261eafb4/internal/metastore/kv/rootcoord/kv_catalog.go#L89-L101

### Expected Behavior

As above code snippet shows, when we tried to remove multiple prefixes of meta, the `maxTxnNum` passed outside was not applied to `etcd.RemoveByBatch`.
In current implementation, if the key has newest timestamp, the snapshot will also save original key without timestamp suffix as a cache to quickly find them. In order to not exceed the txn limit of etcd, we in fact pass the `maxTxnNum` to 64.
In conclusion, above code snippet will finally result in etcd transation error.
